### ISSUE TITLE
Clarifying the Generation of Tensor 'V' 

### ISSUE THREAD

**Issue by euminds** [snap-research/weights2weights#6]
In addition, we observe that the size of the tensor "V" is [99648, 10000], while the tensor produced by combining all LoRA of ID images has a size of [99648, 64974]. I have a few questions regarding this:
Is "V" obtained by performing PCA on the [99648, 64974] tensor?
If so, how is PCA implemented in the code? For example, which specific libraries or functions are used?
If "V" is not obtained through PCA on the [99648, 64974] tensor, what is the process by which it is generated?
Thank you for your assistance!

**Comment by avdravid** [comment]
You are correct. The "V" tensor corresponds to the v matrix when doing SVD/PCA. You can run the following below:

a = torch.load("data.pt")
mean = torch.mean(a, dim=0)
std = torch.std(a, dim=0)
scaled_data = (a - mean)/std
u,s,v = torch.svd(scaled_data)

If you get out-of-memory issues when running SVD on large tensors, you can also use torch.svd_lowrank.
